### PR TITLE
Bug 942365 - fix done button problem in different scenario, r=Arthur

### PR DESCRIPTION
**Pull request by EragonJ**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c68b063971ca33d6f4ffd743de60423827d780af

File: apps/settings/js/panels/simpin_dialog/simpin_dialog.js [hunk 1]
```diff
@@ -493,13 +495,43 @@ define(function(require) {
...
     /**
+     * We have to control `done` button in each condition to make sure
+     * its state is right based on user's input
+     *
+     * @memberOf SimPinDialog
+     * @access private
+     */
+    _updateDoneButtonState: function() {
+      var elements = this._elements;
+      if (this._mode === 'change_pin') {
```
Comment (crh0716) [inline apps/settings/js/panels/simpin_dialog/simpin_dialog.js]: Maybe use switch case?

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=73231e2c91d76afd070faae7471e75a56670eeb4

File: apps/settings/js/panels/simpin_dialog/simpin_dialog.js (hunk 1 above)
Comment (EragonJ) [inline apps/settings/js/panels/simpin_dialog/simpin_dialog.js]: sure okay

**Pull request merged**